docs/README-METRICS.md

```diff
@@ -0,0 +1,165 @@
+actively participates in testing ideas in Firefox.  Key metrics are documented
+below, however other metrics are recorded and analyzed.
+
+## Engagement Ratio
+
+As of this writing, Mozilla has not settled on a final formula for calculating
+an Engagement Ratio, so we'll calculate our ratio based off of [what is
+currently being discussed][2] and change it in the future if needed.  Quoting
+from that discussion:
+
+> Definition of DAU(day): calculate the total number of active users for that
+> date using main_ping.subsessiondate == `day`. Because of data lag, this
+> calculation will not be final until 10 days after `day`.
+>
+> Definition of MAU(day): calculate the total number of active users over a
+> 28-day window `day` - 28 < main_ping.subsessiondate <= `day.
+>
+> On the final dashboard, for each date `E`, display the following calculation:
+> `MEAN(DAU(d) for E - 7 < d <= E) / MAU(E)`
+
+A daily active user will be defined as any user, using their browser that day,
+with at least one test currently enabled in their browser.  This means only
+having the Test Pilot extension installed will *not* count as an active user.
+
+![](metrics-engagement-ratio.png)
+
+## Engagement Breakdown
+
+Within the above Engagement Ratio of a user having at least one test enabled,
+it is interesting to see how many users have how many tests enabled, including
+the negative space of users with Test Pilot installed but with no tests
+enabled.
+
+Total Users here is defined as a Participating User: a user with the Test Pilot
+add-on installed and has run Firefox that day.  They may or may not have any
+tests enabled.
+
+Below is a mockup of what that could look like.
+
+![](metrics-engagement-breakdown.png)
+
+## Service Health
+
+Test Pilot has two critical functions:  firstly to be the broker enabling and
+disabling tests, and secondly, to collect user feedback.  Our main system
+health measures will be the number of tests installed over time and the count
+of user feedback recorded over time.  These will allow us to quickly see
+disruptions in service regardless of cause.
+
+![](metrics-feedback-recorded.png)
+![](metrics-total-installs.png)
+
+# Data Collection Details
+
+There are two sources for metrics data - the server and the client.
+
+**The server** will use server logs to store metric data which will be picked
+up by the measurement team ([some docs][3]), parsed and transformed if
+necessary, and pushed into ElasticSearch (as scale permits), S3, and Redshift.
+Events recorded in the logs will include:
+
+* New user registration
+* New feedback about a test received
+* Test Pilot add-on installed
+* Test installed
+
+Data recorded by the log files will include standard system logging as well as:
+* Test Pilot User ID (if applicable)
+* Test ID (if applicable)
+* Feature Switch Set (if applicable)
+* User Agent
+* Campaign (marketing identifier) (if applicable)
+
+Using the [standard format][4] an example object will look like:
+
+```json
```

Comment:
We have a Python logger that complies with that format in terms of the envelope / metadata, but we don't have anything yet logging that specific `request.summary` type. Probably need to write some Django middleware to do that.

Also, don't I think the `request.summary` type makes sense for these metric events. Seems like `request.summary` is meant to be the equivalent of an Apache `access.log` line generated on every request. 

For the specific metric events listed, we probably want to come up with our own separate type per event - e.g. `testpilot.newuser`, `testpilot.feedback`, `testpilot.main-install`, `testpilot.experiment-install`. Might also make some sense to make the `Fields` closely match whatever we put into Telemetry payloads in case we want to mingle the server / client streams.